source/bayes_simulation.Rmd

```diff
@@ -45,112 +45,171 @@ correctly identifies 80 percent as "OK" but says that 20 percent are
-to be "OK"
+print(f'{k * 100:.2}%')
+```
 
-"31 - 86" = OK, "OK"
+The result is around 5.1%.  If we look at the numbers on our diagram, this makes sense: if we started with 100 cars, 59 are declared OK by a mechanic.  Of those, 3 were faulty, and `3/59 = 0.0508`.
 
-"87 - 100" = OK, "faulty"
+### Probability interpretation
 
-`           REPEAT 1000         `
+**Probability from proportion**
 
-Do 1000 repetitions
+In these examples, it is very common for us to calculate proportions.  In a given simulation: How many cars are declared OK by a mechanic?  59/100.  How many of those 59 were faulty?  3/59.  Given evidence of events, we can count how common they were.
 
-`           GENERATE 1 1,100 a         `
+A natural extension of this idea is to predict the probability of an event occurring.  We can say "in past 100 experiments, condition X occurred 20 times, therefore I  estimate that the probability of X occurring, $P(X)$, is 20/100".  The more trial outcomes we examine, the better our probability estimates become.
 
-Generate a number between "1" and "100"
+**Ratios of proportions / probabilities**
 
-`           IF a between 28 86         `
+At our mechanic's yard, we ask "how many red cars here are broken"?  To calculate that, we first count the number of red cars, then the number of those red cars that are broken, then calculate the ratio: `red_cars_broken / red_cars`.
 
-If it's between "28" and "86" (those that say "good")
+We could just as well have worked in percentages, and we would have gotten the same answer: `percentage_of_red_cars_broken / percentage_of_cars_that_are_red`, since
+that is `(red_cars_broken / 100) / (red_cars / 100)`.  And probabilities would have worked too, given our assumption that the counts are good representations of the yard: `P(red_cars_broken_in_the_yard) / P(red_cars_in_the_yard)`.
 
-`           SCORE a z         `
+**Probability relationships: conditional probability**
 
-Score this number
+We return to the quantity computed above: the **number** of cars being faulty *given* that the mechanic said they were OK.  We use this ratio as an estimate of the probability that that will be true, i.e: the **probability** of cars being faulty *given* that the mechanic said they were OK.
 
-`           END         `
+This is called a *conditional probability*, $P(X | Y)$.  Given a certain outcome Y, calculate the probability of X occurring.
 
-End the IF condition
+Conditional probabilities are calculated as follows:
 
-`           END         `
+$P(X | Y) = P(X \cup Y) / P(Y)$
 
-End REPEAT loop
+From our car example, what is the probability of a car being broken given that a mechanic said it was OK?  It is, out of all the cars that the mechanic said were OK, the fraction that weren't or:
 
-`           COUNT z between 28 30 k         `
+$P(car_broken & mechanic_said_OK) / P(mechanic_said_OK)$.
 
-How many of the SCORED numbers were between "28\--30" (faulty, "OK")
+**Example: conditional probability**
```

Comment:
Well - there's conditional probability, then there's Bayes theorem - which uses conditional probability - Bayes follows pretty quickly from the definition of conditional probability, but it is still a new logical step.